controller/lib/core/sensors.cpp

```diff
@@ -109,30 +110,28 @@ void sensors_init() {
 // Reads a sensor, returning its value in kPa.
 //
 // @TODO: Add alarms if sensor value is out of expected range?
-static float read_pressure_sensor_kpa(Sensor s) {
+static Pressure read_pressure_sensor(Sensor s) {
   int sum = 0;
   for (int i = 0; i < SENSOR_SAMPLES_FOR_READ; i++) {
     sum += Hal.analogRead(pin_for(s)) - sensorZeroVals[s];
   }
   // Sensitivity of all pressure sensors is 1 V/kPa; no division needed.
-  return static_cast<float>(sum) / SENSOR_SAMPLES_FOR_READ * ADC_LSB;
+  return kPa(static_cast<float>(sum) / SENSOR_SAMPLES_FOR_READ * ADC_LSB);
 }
 
-float get_patient_pressure_kpa() {
-  return read_pressure_sensor_kpa(PATIENT_PRESSURE);
+Pressure get_patient_pressure() {
+  return read_pressure_sensor(PATIENT_PRESSURE);
 }
 
-float get_volumetric_inflow_m3ps() {
-  return pressure_delta_to_volumetric_flow(
-      read_pressure_sensor_kpa(INFLOW_PRESSURE_DIFF));
+VolumetricFlow get_volumetric_inflow() {
+  return pressure_delta_to_flow(read_pressure_sensor(INFLOW_PRESSURE_DIFF));
 }
 
-float get_volumetric_outflow_m3ps() {
-  return pressure_delta_to_volumetric_flow(
-      read_pressure_sensor_kpa(OUTFLOW_PRESSURE_DIFF));
+VolumetricFlow get_volumetric_outflow() {
+  return pressure_delta_to_flow(read_pressure_sensor(OUTFLOW_PRESSURE_DIFF));
 }
 
-float pressure_delta_to_volumetric_flow(float diffPressureInKiloPascals) {
+VolumetricFlow pressure_delta_to_flow(Pressure delta) {
```

Comment:
Maybe rename the variables with units too.

Reply:
Here this becomes pretty unfortunate, because you have

```
bernouliAreaDivisor_sqrt_inverse_m2
```

or something.  Since these never leave the short function, I don't think it's a win.